test/unit/core/http/http_error_test.rb

```diff
@@ -17,7 +17,7 @@
 
 describe Azure::Core::Http::HTTPError do
   let :http_response do
-    stub(body: Azure::Core::Fixtures[:http_error], status_code: 409, uri: 'http://dummy.uri', headers: {})
+    stub(body: Azure::Core::Fixtures[:http_error], status_code: 409, uri: 'http://dummy.uri', headers: { 'Content-Type' => 'application/atom+xml' })
```

Comment:
Would you mind adding the tests around `json` responses and `unknown` responses?

Reply:
Sure, the code has been updated and added tests.

Reply:
👍